Project 5 For John Beverly

```diff
@@ -51,3 +51,441 @@ You may find the following resources useful:
+To use this constraint in this was one would start the constraint with something like 
+
+c.	a -> b ^ b -> a
+
+ex:TwitterFollowersSymmetryShape and follow the general format of this constraint. 
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:SymmetryShape
+
+a sh:NodeShape ;
+
+sh:targetclass ex:relation ;
+
+sh:property [    
+
+sh:path rdf:value ;    
+
+sh:minCount 2 ;   
+
+sh:maxCount 8 ;
+
+sh:datatype xsd:string ;    
+
+sh:nodeKind sh:Literal ;    
+
+sh:message "The relation does not have symmetric values"  
+
+] ;
+
+sh:property [    
+
+sh:path [ 
+
+sh:inversePath rdf:value 
+
+] ;
+
+sh:minCount 2 ;
+
+sh:datatype xsd:string ;
+
+sh:nodeKind sh:Literal ;
+
+sh:message "The relation does not have symmetric values"
+
+] .
+
+
+
+4.	Constraint for Reflexivity 
+a.	Our constraint checks for reflexivity in a graph.
+
+b.	An example of when one would use this constraint would be if we had a graph that represented published philosophers to check for when philosophers are self-referencing their own work in a novel piece. Such that when a philosopher published novel work that references older work this constraint would alert to the self-reference. One would change ex:ReflexivityShape to ex:PhilosopherReferenceReflexivityShape and follow the format of the constraint below.  
+
+c.	a -> a
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+
+ex:ReflexiveShape
+
+a sh:NodeShape ;
+
+sh:targetClass ex:relation ;
+
+sh:property [    
+
+sh:path rdf:value ;    
+
+sh:minCount 1 ;    
+
+sh:maxCount 6 ;    
+
+sh:datatype xsd:string ; 
+
+] .
+
+5.	Constraint for Irreflexive 
+a.	Our constraint checks for irreflexively in a graph 
+
+b.	An example of when to use this constraint would be if you had a graph that represented a company’s organizational structure. Such that no employee should be their own manager. This constraint would ensure that is not possible. One would simply change ex:IrrelexiveShape to ex:CompanyIrrelexiveOrganizationShape and follow the formatting below. 
+
+c.	~(a -> a)
+
+@prefix sh: <http://www.w3.org/ns/shacl#> .
+@prefix ex: <http://example.com/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
+
+ex:IrreflexiveShape
+
+a sh:NodeShape ;
+
+sh:targetClass ex:relation ;
+
+sh:property [    
+
+sh:path rdf:value ;    
+
+sh:minCount 1 ;    
+
+
+sh:maxCount 4 ;    
+
+sh:datatype xsd:string ; 
+
```

Comment:
From my perspective, it seems that what really play a role in your shapes are sh:minCount and sh:maxCount, is this right? Sorry I cannot find other constraints.